<a href="https://colab.research.google.com/github/stepkurniawan/network-analysis-wikipedia-journals/blob/main/3_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Analysis
To compare the dataset with the original paper


# Data Cleaning

In [5]:
import networkx as nx
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
plt.style.use('seaborn')
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')
import itertools

In [6]:
#run this for google colab
# ! git clone https://github.com/stepkurniawan/network-analysis-wikipedia-journals.git
# df_journal = pd.read_csv('/content/network-analysis-wikipedia-journals/files2/df_journal.csv')
# df_num_citations = pd.read_csv('/content/network-analysis-wikipedia-journals/files2/df_num_citations.csv')
# dfnew = pd.read_csv('/content/dfnew.csv') # big df from data cleaning

# Data Prep

In [7]:
# # run this for local
# # df_journal = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files\\df_journal.csv')
# df_journal = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_journal.csv')
# df_num_citations = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_num_citations.csv')
# df_node_full = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_node_full.csv')
# df_trim_edges = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_trim_edges.csv')
# df_node_size_trim = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_node_size_trim.csv')


In [9]:
# Stephen: please dont change, you can comment
# df_journal = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_journal.csv')
# df_journal = pd.read_csv('C:\\Users\\steph\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_journal.csv')

df_journal = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_journal.csv')
df_num_citations = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_num_citations.csv')
df_node_full = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_node_full.csv')
df_trim_edges = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_trim_edges.csv')
df_node_size_trim = pd.read_csv('D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_node_size_trim.csv')

import glob
import os

df_final_path = r'D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_final'
df_final_all_files = glob.glob(os.path.join(df_final_path , "*.csv"))
df_final_all_files

li = []
for df_final_filename in df_final_all_files:
    df_final = pd.read_csv(df_final_filename, index_col=None)
    li.append(df_final)
df_final = pd.concat(li, axis=0, ignore_index=True)

df_edges_full_path = r'D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_edges_full'
df_edges_full_all_files = glob.glob(os.path.join(df_edges_full_path , "*.csv"))
df_edges_full_all_files

li = []
for df_edges_full_filename in df_edges_full_all_files:
    df_edges_full = pd.read_csv(df_edges_full_filename, index_col=None)
    li.append(df_edges_full)
df_edges_full = pd.concat(li, axis=0, ignore_index=True)


df_article_path = r'D:\\OneDrive - Leuphana Universität\\Semester 2\\4 - Network Analysis\\Second Group Work\\network-analysis-wikipedia-journals\\files2\\df_article'
df_article_all_files = glob.glob(os.path.join(df_article_path , "*.csv"))
df_article_all_files

li = []
for df_article_filename in df_article_all_files:
    df_article = pd.read_csv(df_article_filename, index_col=None)
    li.append(df_article)
df_article = pd.concat(li, axis=0, ignore_index=True)




In [17]:
df_final

Unnamed: 0                                            Authors  \
0              3777  [{last=Alley, first=Richard B.}, {last=JC Stra...   
1              3778  [{link=Joel Mokyr, last=Mokyr, first=Joel\n}, ...   
2              3779  [{link=Joel Mokyr, last=Mokyr, first=Joel\n}, ...   
3              3780                                                NaN   
4              3781                         [{last=Falany, first=CN.}]   
...             ...                                                ...   
1996716     5857817                               [{last=V. P. Dimri}]   
1996717     5857818  [{last=VanderLaan, first=Doug P.}, {last=Ren, ...   
1996718     5857819                     [{last=White, first=Patricia}]   
1996719     5857820                 [{last=Whitfield, first=James B.}]   
1996720     5857821  [{last=Wickramasinghe, first=Nethu}, {last=Cha...   

                                                   ID_list  \
0        {ISBN=978-0-8137-2337-2, DOI=10.1130/0-8137-23...   
1               {DOI=10.1111/j.1468-0289.1984.tb00344.x\n}   
2               {DOI=10.1111/j.1468-0289.1984.tb00344.x\n}   
3                                                      NaN   
4          {PMID=9068609, DOI=10.1096/fasebj.11.4.9068609}   
...                                                    ...   
1996716                                                NaN   
1996717     {PMID=24091924, DOI=10.1007/s12110-013-9182-z}   
1996718                     {DOI=10.1215/02705346-2016996}   
1996719  {ISBN=978-0-12-031754-7, PMID=14711084, DOI=10...   
1996720                    {DOI=10.11646/zootaxa.3620.1.5}   

                                                     Title  \
0        Glacial processes, past and present: Glaciaolo...   
1        New Developments in Irish Population History, ...   
2        New Developments in Irish Population History, ...   
3                  The True Confessions of Charlotte Doyle   
4          Enzymology of human cytosolic sulfotransferases   
...                                                    ...   
1996716           Pramod Sadasheo Moharir (1943\u20132003)   
1996717  Male androphilia in the ancestral environment....   
1996718  Looking Back and Forward: A Conversation about...   
1996719  Phylogenetic Insights into the Evolution of Pa...   
1996720  Lost and found: One of the world's most elusiv...   

                                                 citations        id  \
0        {{Cite journal  | last = Alley  | first =  Ric...  30112151   
1        {{Cite journal  | last = Mokyr  | first = Joel...  11693628   
2        {{Cite journal  | last = Mokyr  | first = Joel...   1141960   
3        {{Cite journal  | last = | first = | title = T...   3057743   
4        {{Cite journal  | last1 = Falany | first1 = CN...  29199666   
...                                                    ...       ...   
1996716  {{Cite journal |last=V. P. Dimri |title=Pramod...     37533   
1996717  {{Cite journal |last=VanderLaan |first=Doug P....  20756869   
1996718  {{Cite journal |last=White |first=Patricia |da...  48871297   
1996719  {{Cite journal |last=Whitfield |first=James B....   5457188   
1996720  {{Cite journal |last=Wickramasinghe |first=Net...  12411662   

                                      page_title  \
0                                  overdeepening   
1                                      ballitore   
2                                       straffan   
3        the true confessions of charlotte doyle   
4                  carbohydrate sulfotransferase   
...                                          ...   
1996716              indian institute of science   
1996717                              eusociality   
1996718                            women in film   
1996719                          parasitoid wasp   
1996720                pseudophilautus stellatus   

                         PublisherName                 journal_name  
0        Geological Society of America            special paper 337  


Unnamed: 0       1996721
Authors          1293850
ID_list          1105285
Title            1492379
citations        1815931
id                486799
page_title        486710
PublisherName      32687
journal_name      176515
dtype: int64

In [23]:
# df_new = pd.read_csv('C:\\Users\\Beka\Documents\\GitHub\\network-analysis-wikipedia-journals\\files2\\df_new.csv')
# df_new

In [7]:
df_journal

page_title  \
0                                     overdeepening   
1                                         ballitore   
2                                          straffan   
3           the true confessions of charlotte doyle   
4                     carbohydrate sulfotransferase   
...                                             ...   
1996716                          muscle contraction   
1996717                                darrell issa   
1996718                                seth rollins   
1996719                                   chia seed   
1996720  potential applications of carbon nanotubes   

                                            journal_name  
0                                      special paper 337  
1                            the economic history review  
2                            the economic history review  
3                                 the horn book magazine  
4                                                faseb j  
...                                                  ...  
1996716                                  br j sports med  
1996717                     open medicine john willinsky  
1996718                        pro wrestling illustrated  
1996719                                        nutr hosp  
1996720  journal of biomedical materials research part a  

[1996721 rows x 2 columns]

In [8]:
df_journal.nunique()

page_title      486710
journal_name    176515
dtype: int64

In [9]:
df_num_citations

journal_name  num_citation
0                                nature         35475
1       journal of biological chemistry         30200
2                                  pnas         29544
3                               science         25924
4                              plos one         12602
...                                 ...           ...
176510         bull inst fr afr noire a             1
176511    httpwwwluxurytraveladvisorcom             1
176512   httpwwwluxurytravelmagazinecom             1
176513                 newcastle herald             1
176514        overland literary journal             1

[176515 rows x 2 columns]

In [10]:
df_num_citations = df_journal[['journal_name']].value_counts().reset_index(name='num_citation')
df_num_citations.head(10)

journal_name  num_citation
0                                   nature         35475
1          journal of biological chemistry         30200
2                                     pnas         29544
3                                  science         25924
4                                 plos one         12602
5                                billboard         10662
6  the iucn red list of threatened species         10108
7                                     cell          9189
8                                  zootaxa          8241
9                              lloyds list          7347

In [11]:
df_num_citations.nunique()

journal_name    176515
num_citation       818
dtype: int64

In [12]:
df_node_full # 176 656 rows

ID  weight   area
0                                nature   35475  Multi
1       journal of biological chemistry   30200     LS
2                                  pnas   29544  Multi
3                               science   25924  Multi
4                              plos one   12602  Multi
...                                 ...     ...    ...
176651         bull inst fr afr noire a       1    NaN
176652    httpwwwluxurytraveladvisorcom       1    NaN
176653   httpwwwluxurytravelmagazinecom       1    NaN
176654                 newcastle herald       1    NaN
176655        overland literary journal       1    NaN

[176656 rows x 3 columns]

In [13]:
df_trim_edges

source  \
0                                          nature   
1                                          nature   
2                 journal of biological chemistry   
3                                            pnas   
4                 journal of biological chemistry   
..                                            ...   
636                               russian history   
637                                      plos one   
638                                all about soap   
639                           cretaceous research   
640  journal of personality and social psychology   

                                         target  weight  
0                                       science   51706  
1                                          pnas   43631  
2                                          pnas   40297  
3                                       science   33636  
4                                        nature   24171  
..                                          ...     ...  
636                              russian review    1406  
637                  quaternary science reviews    1406  
638                                 inside soap    1405  
639    journal of south american earth sciences    1401  
640  personality and social psychology bulletin    1400  

[641 rows x 3 columns]

In [14]:
df_node_size_trim

ID  size     area
0                        acta palaeontologica polonica  1316       PS
1                          acta palaeontologica sinica    87      NaN
2    alcheringa an australasian journal of palaeont...   339       PS
3                                       all about soap   226      NaN
4                                          ameghiniana   305  LS & PS
..                                                 ...   ...      ...
232                                          wwwdsmzde  1445      NaN
233                                      wwwuniprotorg  3133      NaN
234                                            zookeys  2266       LS
235          zoological journal of the linnean society  3304       LS
236                                            zootaxa  8241       LS

[237 rows x 3 columns]

# Answering questio

In [ ]:
#1. 14K journals -> how many article in 1 journal? Mu and sigma 


In [21]:
#2. 1 journal how many citations? (page title) (mu and sigma)
print(df_num_citations.mean())
print(df_num_citations.std())



num_citation    11.283489
dtype: float64
num_citation    174.545585
dtype: float64


In [20]:
# 3. 1-page title, how many journals? (mu and sigma)

df_pivot = df_journal.groupby('page_title')[['journal_name']].count()
print (df_pivot.mean())
print (df_pivot.std())

journal_name    4.09218
dtype: float64
journal_name    8.983258
dtype: float64


In [ ]:
# 3.

In [17]:
# 2. simple mean and std of journals cited

df_num_citations.nunique()
print('mean of num_citation', df_num_citations.mean()) # 8.55975
print('standard deviation of num_citation', df_num_citations.std()) # 104.867143

mean of num_citation num_citation    11.283489
dtype: float64
standard deviation of num_citation num_citation    174.545585
dtype: float64


# SCOPUS

In [25]:
# 2. which areas are cited more? 

# full
df_node_full.groupby(["area"]).sum().sort_values(["weight"], ascending = False)


weight
area                
LS            300402
PS            173621
SSH           161138
HS            150308
LS & HS       120282
Multi         114319
LS & PS        59844
SSH & PS       22615
SSH & HS       19092
LS & SSH        9172
LS, PS & HS     8612
PS & HS         5164
LS, SSH & HS    2764
LS, SSH & PS    2732
SSH, PS & HS    1110

In [29]:
# trim
df_node_size_trim.groupby(["area"]).sum().sort_values(["size"], ascending = False)


size
area                
LS            153994
Multi         111226
PS             61035
LS & HS        43528
HS             28191
LS & PS        18685
SSH             9027
LS, PS & HS     1579
SSH & HS        1039
LS & SSH         971
LS, SSH & PS     712
SSH & PS         641
PS & HS          184

In [13]:
# 5. Mean of each area
# full
df_node_full.groupby(["area"]).mean().sort_values(["weight"], ascending = False)

weight
area                     
Multi         1524.253333
LS             143.870690
LS & HS         79.604236
LS & PS         74.156134
LS & SSH        73.967742
LS, PS & HS     55.561290
PS              42.721703
LS, SSH & PS    42.030769
LS, SSH & HS    38.388889
SSH & HS        35.486989
HS              33.784671
SSH             27.859267
PS & HS         27.763441
SSH, PS & HS    24.666667
SSH & PS        21.579198

In [31]:
# trim
df_node_size_trim.groupby(["area"]).mean().sort_values(["size"], ascending = False)

size
area                      
Multi         11122.600000
LS             2081.000000
HS             2013.642857
LS & HS        1741.120000
LS, PS & HS    1579.000000
PS             1298.617021
SSH & HS       1039.000000
LS & PS        1038.055556
LS & SSH        971.000000
LS, SSH & PS    712.000000
SSH             644.785714
SSH & PS        320.500000
PS & HS         184.000000

# General Discussion

Compared to the original paper that analyzed only 847.512 references to scientific articles distributed across 193.802 Wikipedia entries from the year 2018,

we have: 1.997.167 references, across 176515 Wikipedia entries from the year 2020. 


In [4]:
df_node_full["weight"].sum()

1997167

In [19]:
# how many citation / references ? 1 996 721
# how many unique articles cited? 1 492 379
# how many wikipedia page? 486 710
# how many journals cited? 176 515
df_final.nunique()

Unnamed: 0       1996721
Authors          1293850
ID_list          1105285
Title            1492379
citations        1815931
id                486799
page_title        486710
PublisherName      32687
journal_name      176515
dtype: int64

In [8]:
# each wiki entry includes xx references to articles

df_wikipedia_vs_article = df_final.groupby('page_title', as_index=False)['Title'].count()
print(df_wikipedia_vs_article["Title"].mean()) # 4.090608370487559

print(df_wikipedia_vs_article["Title"].std()) # 8.983430842161379
df_wikipedia_vs_article


4.090608370487559
8.983430842161379


page_title  Title
0                                  !women art revolution      3
1                                       "a" is for alibi      1
2       "as the old sing, so pipe the young" (jan steen)      2
3                                "baby lollipops" murder      3
4                             "believing women" in islam      1
...                                                  ...    ...
486705                                          ‘aziziya      1
486706                                          ‘elepaio      4
486707                                         ‘eua rail      1
486708                                     ‘ofa likiliki      3
486709                                                ♯p      1

[486710 rows x 2 columns]

In [25]:
# how many article receive only one citation?  1.270.531  from  1.996.721 = 63.63%

df_wikipedia_vs_article_one_cit = df_article.groupby('Title', as_index=False)['page_title'].count()
# df_wikipedia_vs_article_one_cit= df_wikipedia_vs_article_one_cit.sort_values("Title", ascending = False)
df_wikipedia_vs_article_one_cit = df_wikipedia_vs_article_one_cit[df_wikipedia_vs_article_one_cit["page_title"] == 1]
df_wikipedia_vs_article_one_cit

Title  page_title
0                                      The First Part Last           1
1                                                       ()           1
3         (A = Li, Na): Crystallization of a Rhodium(V)...           1
4         (M=Co, Ni): Crystal Structure and IR Spectros...           1
5         -A A +A Industry Watch: Chinese Lead Immigran...           1
...                                                    ...         ...
1492374   'Malbrough s'en va-t-en guerre': Seven Canadi...           1
1492375          Sciurillus pusillus (Rodentia: Sciuridae)           1
1492376  – – – bror av Vilhelm. Omkring Thomas P. Krag ...           1
1492377                                — From Fins to Feet           1
1492378  ’Presencia’ afrouruguaya en Buenos Aires: Su i...           1

[1270531 rows x 2 columns]

In [21]:
# a total of 1.997.167  journals are cited through wikipedia references
# top 10 most cited journals in wikipedia

print(df_node_full["weight"].sum())

df_node_full.head(10)

1997167


ID  weight   area
0                                   nature   35475  Multi
1          journal of biological chemistry   30200     LS
2                                     pnas   29544  Multi
3                                  science   25924  Multi
4                                 plos one   12602  Multi
5                                billboard   10662    NaN
6  the iucn red list of threatened species   10108    NaN
7                                     cell    9189     LS
8                                  zootaxa    8241     LS
9                              lloyds list    7347    NaN

In [15]:
# number of co-citation in the network 22.717.803
df_edges_full["weight"].sum()

22717803

In [18]:
df_edges_full.head(10)
# with a high number of self-citation from lloyds list, billboard, new lloyds list, and journal of biological chemistry. 

source                           target  weight
0                      lloyds list                      lloyds list  314693
1                        billboard                        billboard   97850
2                  new lloyds list                  new lloyds list   91271
3  journal of biological chemistry  journal of biological chemistry   61268
4                           nature                          science   51706
5                           nature                             pnas   43631
6                 gaceta de madrid                 gaceta de madrid   41051
7                           nature                           nature   40593
8  journal of biological chemistry                             pnas   40297
9              cretaceous research              cretaceous research   34379